ublox/src/ubx_packets/packets.rs

```diff
@@ -1970,15 +1972,40 @@ struct RxmRtcm {
 }
 
 #[ubx_packet_recv]
-#[ubx(class = 0x10, id = 0x02, fixed_payload_len = 16)]
+#[ubx(class = 0x10, id = 0x02, max_payload_len = 1240)]
 struct EsfMeas {
     time_tag: u32,
+    #[ubx(map_type = EsfMeasInfo, from = EsfMeasInfo::new)]
+    info: [u8; 0],
+}
+
+#[derive(Debug)]
+pub struct EsfMeasInfo {
     flags: u16,
     id: u16,
-    /// flags >> 11 & 0x1F
-    data: u32,
-    /// flags & 0x8
-    calib_tag: u32,
+    data: Vec<u32>,
```

Comment:
@notoriaga wanted to ask, same with before, is it just overall better to keep as Iterator rather than Vec? - is it overall worse to keep as `vec` since that's what i've been storing things in icbins with structs... (only read that needs to be deallocated / slices won't etc etc)

Reply:
There really isn't a better or worse, and sometimes you'll use both. A `Vec` resizable array that stores its contents in the heap. An `Iterator` is more generic than that. The stuff it's yielding can be something on the heap (like if you create an iterator from `vec.iter()`), it could be something on the stack (like from `array.iter()`), or it could not even be referencing some underlying data source (like [this](https://doc.rust-lang.org/std/iter/fn.successors.html) which generates data to be returned)

Really anything can be an iterator (you just have to implement `Iterator` for it), whereas `Vec` is a specific container.

In our case here we kind of need to use an iterator, because a no_std/no alloc crate has no Vec. You could maybe do something with arrays but iterators solve this problem well (and can be turned into an array or vec if the user wants)